### PR TITLE
Add Base64 encoding/decoding snippets

### PR DESCRIPTION
**Pull request by swarajsaaj**
Add Base64 encoding/decoding snippets in a new package "encoding"
- Base64EncodeSnippet
- Base64DecodeSnippet

### CODE REVIEW

File: src/main/java/encoding/Base64DecodeSnippet.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/*
+ * 30 Seconds of Java code library
+ *
+ */
+public class Base64DecodeSnippet {
+  /**
+   * Decodes a Base64 encoded string to the actual representation.
+   *
+   * @param input The base64 encoded string
```
Comment (Ascenio) [inline src/main/java/encoding/Base64DecodeSnippet.java:37]:
```java
@param input base64 encoded string
```

File: src/test/java/encoding/Base64EncodeSnippetTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class Base64EncodeSnippetTest {
+  /**
+   * Tests for {@link Base64EncodeSnippet#encodeBase64(String)}.
+   */
+  @Test
+  void testBase64Encoding() {
+    var encodedString = Base64EncodeSnippet.encodeBase64("TestString");
+    assertEquals("VGVzdFN0cmluZw==",encodedString);
+  }
+}
```
Comment (Ascenio) [inline src/test/java/encoding/Base64EncodeSnippetTest.java:44]: Should add a EOL

File: src/test/java/encoding/Base64DecodeSnippetTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+class Base64DecodeSnippetTest {
+  /**
+   * Tests for {@link Base64DecodeSnippet#decodeBase64(String)}.
+   */
+  @Test
+  void testBase64Decoding() {
+    var decodedString = Base64DecodeSnippet.decodeBase64("VGVzdFN0cmluZw==");
+    assertEquals("TestString",decodedString);
+  }
+}
```
Comment (Ascenio) [inline src/test/java/encoding/Base64DecodeSnippetTest.java:44]: Should add a EOL

File: src/main/java/encoding/Base64EncodeSnippet.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+/*
+ * 30 Seconds of Java code library
+ *
+ */
+public class Base64EncodeSnippet {
+  /**
+   * Encodes the input string to a Base64 encoded string.
+   *
+   * @param input The string to be encoded.
```
Comment (Ascenio) [inline src/main/java/encoding/Base64EncodeSnippet.java:41]:
```java
@param input the string to be encoded
```

File: src/main/java/encoding/Base64EncodeSnippet.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+   * @return base64 encoded string of the input
```
Comment (Ascenio) [inline src/main/java/encoding/Base64EncodeSnippet.java:42]:
```java
@return base64 encoded string
```

File: src/main/java/encoding/Base64DecodeSnippet.java (hunk 1 above)
Comment (swarajsaaj) [reply src/main/java/encoding/Base64DecodeSnippet.java]: Fixed in new commit.

File: src/test/java/encoding/Base64EncodeSnippetTest.java (hunk 2 above)
Comment (swarajsaaj) [reply src/test/java/encoding/Base64EncodeSnippetTest.java]: Fixed in new commit.

File: src/main/java/encoding/Base64EncodeSnippet.java (hunk 4 above)
Comment (swarajsaaj) [reply src/main/java/encoding/Base64EncodeSnippet.java]: Fixed in new commit.

File: src/main/java/encoding/Base64EncodeSnippet.java (hunk 5 above)
Comment (swarajsaaj) [reply src/main/java/encoding/Base64EncodeSnippet.java]: Fixed in new commit.

**Comment by swarajsaaj** [comment]
Thanks for review @Ascenio . The comments have been fixed in a new commit.

File: src/test/java/encoding/Base64DecodeSnippetTest.java (hunk 3 above)
Comment (swarajsaaj) [reply src/test/java/encoding/Base64DecodeSnippetTest.java]: Added.

**Review by xdvrx1 — APPROVED** [review]
I think this one is good to go,

File: src/main/java/encoding/Base64EncodeSnippet.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+ * SOFTWARE.
+ */
+
+package encoding;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.stream.Collectors;
```
Comment (iluwatar) [inline src/main/java/encoding/Base64EncodeSnippet.java:31]: Please remove the unused imports

File: src/test/java/encoding/Base64DecodeSnippetTest.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ */
+class Base64DecodeSnippetTest {
+  /**
+   * Tests for {@link Base64DecodeSnippet#decodeBase64(String)}.
+   */
+  @Test
+  void testBase64Decoding() {
+    var decodedString = Base64DecodeSnippet.decodeBase64("VGVzdFN0cmluZw==");
+    assertEquals("TestString",decodedString);
```
Comment (iluwatar) [inline src/test/java/encoding/Base64DecodeSnippetTest.java:42]: In addition to the happy path it's good to test the negative case and for example what happens when you call the method with null or empty parameter? Similarly for the encoding test.

**Review by iluwatar — CHANGES_REQUESTED** [review]

File: src/test/java/encoding/Base64DecodeSnippetTest.java (hunk 7 above)
Comment (swarajsaaj) [reply src/test/java/encoding/Base64DecodeSnippetTest.java:42]: Added another tests with empty inputs

File: src/main/java/encoding/Base64EncodeSnippet.java (hunk 6 above)
Comment (swarajsaaj) [reply src/main/java/encoding/Base64EncodeSnippet.java]: Fixed

**Review by iluwatar — APPROVED** [review]

**Comment by iluwatar** [comment]
Great contribution @swarajsaaj 👍🏻 

Many thanks for reviewers @Ascenio and @xdvrx1 as well
